roqoqo/src/noise_models/overrotation.rs

```diff
@@ -0,0 +1,442 @@
+// Copyright © 2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::SupportedVersion;
+use std::collections::HashMap;
+/// Description of single qubit overrotation noise model, [roqoqo::noise_models::SingleQubitOverrotationOnGate].
+///
+/// Consists of the raw data needed to construct a rotation gate that adds
+/// overrotation: gate name and statistics (mean and standard deviation) of a Gaussian distribution
+/// from which the overrotation angle is sampled.
+/// Example:
+///
+/// ```
+/// use roqoqo::noise_models::SingleQubitOverrotationDescription;
+/// let gate = "RotateX";
+/// let theta_mean = 0.0;
+/// let theta_std = 1.0;
+/// let mut noise_desc = SingleQubitOverrotationDescription::new(gate, theta_mean, theta_std);
+/// ```
+#[derive(Debug, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "serialize",
+    serde(from = "SingleQubitOverrotationDescriptionSerialize")
+)]
+#[cfg_attr(
+    feature = "serialize",
+    serde(into = "SingleQubitOverrotationDescriptionSerialize")
+)]
+pub struct SingleQubitOverrotationDescription {
+    /// Name of the single qubit rotation gate
+    gate: String,
+    /// Mean value for the overroation: overroations are stochatically distributed around this base overroation value
+    theta_mean: f64,
+    /// Standard deviation of Gaussian distribution around mean value of theta
+    theta_std: f64,
+}
+
+#[cfg(feature = "json_schema")]
+impl schemars::JsonSchema for SingleQubitOverrotationDescription {
+    fn schema_name() -> String {
+        "SingleQubitOverrotationDescription".to_string()
+    }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        <SingleQubitOverrotationDescriptionSerialize>::json_schema(gen)
+    }
+}
+
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Out of interest, is this custom serialisation required here? It seems it uses the same fields as the normal struct, so I don't see what this brings.